### PR TITLE
refactor the github actions configuration

### PR DESCRIPTION
**Pull request by devoncarew**
Refactor the github actions configuration:
- use a matrix setup to test across flutter versions (master, latest) and packages in the repo
- run `dart fix --apply` to address ~5 API deprecations
- exclude `visibility_detector` on stable (we'll want to re-enable this after the next stable flutter is released)

### CODE REVIEW

**Comment by devoncarew** [comment]
cc @jamesderlin; this removes the older script, and shards testing across the 5 packages and 2 release channels of flutter

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -4,18 +4,39 @@ on:
...
   build:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        package: [flutter_simple_treeview, linked_scroll_controller, scrollable_positioned_list, self_storing_input, visibility_detector]
+        version: [stable, master]
+        exclude:
+          - package: visibility_detector
+            version: stable
```
Comment (xster) [inline .github/workflows/build.yaml:22]: wonder what this does exactly. If the pubspec constraints for that package is higher than the current stable, it'll just fail pub get?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -4,18 +4,39 @@ on:
...
+  pull_request:
+    branches: [ master ]
 
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        package: [flutter_simple_treeview, linked_scroll_controller, scrollable_positioned_list, self_storing_input, visibility_detector]
```
[the remark this answers is not in the document]
Comment (devoncarew) [reply .github/workflows/build.yaml:18]: This is definitely possible; one way would be to have a small script (in `tool/` ?), which compares the list here with the packages found in `packages/`, and fail the CI if there's a difference.

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -4,18 +4,39 @@ on:
...
+          channel: ${{ matrix.version }}
+
+      - name: flutter version
+        run: flutter --version
+
+      - name: flutter pub get
+        run: flutter pub get
+        working-directory: packages/${{ matrix.package }}
+
+      - name: flutter analyze
```
[the remark this answers is not in the document]
Comment (devoncarew) [reply .github/workflows/build.yaml:36]: added in 952a86b54448622451d00fd67b5f01926107e2bd

**Review by jamesderlin — APPROVED** [review]
LGTM, aside from not checking that all packages are covered.  That can be done separately if you prefer, though.

**Comment by devoncarew** [comment]
> LGTM, aside from not checking that all packages are covered. That can be done separately if you prefer, though.

Thanks for the review! I think the check is valuable, but would prefer to cover that in a separate PR (this PR as is already increases coverage, and makes the CI output easier to figure out).

**Comment by devoncarew** [comment]
> checking that all packages are covered

Done in https://github.com/google/flutter.widgets/pull/382 -

**Pull request merged**